holmes/core/tool_calling_llm.py

```diff
@@ -357,6 +365,97 @@ def truncate_messages_to_fit_context(
                 message["content"] = message["content"][:tool_size]
         return messages
 
+    def call_stream(
+        self,
+        system_prompt: str,
+        user_prompt: Optional[str] = None,
+        response_format: Optional[Union[dict, Type[BaseModel]]] = None,
+        sections: Optional[InputSectionsDataType] = None,
+    ):
+        messages = [
```

Comment:
Since most of this code is duplicate, there isn't a way to refactor and add only streaming support to the original function? or refactor some parts of it and reuse?

Reply:
added as a technical debt for next sprint to arrange these function .